Scripts/EllosBoonSelectorMod.lua

```diff
@@ -609,6 +674,79 @@ function CloseSeedControlScreen( screen, button )
   OnScreenClosed({ Flag = screen.Name })
 end
 
+function IsSecondRoomRewardEligible(requirements, firstRoomReward)
+  if requirements.RequiredUpgradeableGodTraits ~= nil and requirements.RequiredUpgradeableGodTraits >= 1 and firstRoomReward ~= "Boon" then
+    return false
+  end
+  if requirements.RequiredMaxWeaponUpgrades ~= nil and requirements.RequiredMaxWeaponUpgrades < 1 and firstRoomReward == "Hammer" then
+    return false
+  end
+  if requirements.RequiredMinDepth ~= nil and requirements.RequiredMinDepth >= 2 then
+    return false
+  end
+  if requirements.RequiredMinBiomeDepth ~= nil and requirements.RequiredMinBiomeDepth >= 2 then
+    return false
+  end
+  if requirements.RequiredFalseCosmetics ~= nil then
+    return false
+  end
+  return true
+end
+
+function PredictSecondRoomReward(seedForPrediction, firstRoomShrine, firstRoomReward)
+  RandomSetNextInitSeed( {Seed = seedForPrediction} )
+  local rewardStore = "RunProgress"
+  if not firstRoomShrine then
+    RandomSynchronize(1) -- Known offset at which the RNG rolls the room
+    rewardStore = GetRandomValue({
+      "MetaProgress",
+      "MetaProgress",
+      "MetaProgress",
+      "MetaProgress",
+      "MetaProgress",
+      "MetaProgress",
+      "MetaProgress",
+      "MetaProgress",
+      "MetaProgress",
+      "MetaProgress",
+      "MetaProgress",
+      "MetaProgress",
+      "MetaProgress",
+      "MetaProgress",
+      "MetaProgress",
+      "MetaProgress",
+      "MetaProgress",
+      "MetaProgress",
+      "RunProgress" -- Athena room
```

Comment:
lol nice.  Maybe leave a comment saying this is brittle if the number of eligible 2nd rooms change.  I'm just thinking about compatibility with other mods (e.g. a room randomizer).